app/services/hashed_password.rb

```diff
@@ -0,0 +1,75 @@
+# HashedPassword is returned from PasswordHasher.hash.
+# HashedPassword outputs various key hashing related to NIST encryption.
+#
+# Generate and store a 128-bit salt S.
+# Z1, Z2 = scrypt(S, password)   # split 256-bit output into two halves
+# Generate random R.
+# D = KMS_GCM_Encrypt(key=server_secret, plaintext=R) xor Z1
+# E = hash( Z2 + R )
+# F = hash(E)
+# Store F (User.encrypted_password) and D (User.encryption_key) in db
+
+class HashedPassword
```

Comment:
I was going mostly for parity with `PasswordHasher`:
```ruby
hashed_password = PasswordHasher.hash(password)
```

`HashedPassword` does a lot of things, mostly because the NIST model does a lot of things and I eventually found it helpful to keep most of the NIST-specific actions in a single class. Maybe we could call it `NistPassword`?